### PR TITLE
Wip/mehrad/fixes

### PR DESCRIPTION
**Pull request by Mehrad0711**
Changes:
- Cache any transformer embedding (so we can use bart and mbart)
- Log context during prediction

### CODE REVIEW

File: genienlp/data_utils/embeddings.py [hunk 1]
```diff
@@ -214,7 +213,7 @@ def load_embeddings(cachedir, context_emb_names, question_emb_names, decoder_emb
             continue
 
         emb_type = get_embedding_type(emb_name)
-        if _is_bert(emb_name) or _is_xlmr(emb_name):
+        if emb_name in ALL_PRETRAINED_MODEL_ARCHIVE_MAP.keys():
```
Comment (gcampax) [inline genienlp/data_utils/embeddings.py:216]: Don't you need Tokenizer subclasses for new models?

Comment (Mehrad0711) [reply genienlp/data_utils/embeddings.py:216]: So this is meant to be used with "cache-embeddings" command to cache models locally or for the docker. The model and tokenizer will be instantiated from "run_lm_finetune" and "run_generation" script.

Comment (gcampax) [reply genienlp/data_utils/embeddings.py:216]: Yeah then we don't need it, because the docker doesn't cache the pretrained models anymore.

Comment (Mehrad0711) [reply genienlp/data_utils/embeddings.py:216]: Mbart models are pretty big and it's better to cache them (in our own dockers). It speeds up subsequent training.

Comment (gcampax) [reply genienlp/data_utils/embeddings.py:216]: You still have to download the docker image on each machine for each training. The difference is not that big...

Comment (Mehrad0711) [reply genienlp/data_utils/embeddings.py:216]: Agreed

**Pull request closed without merging**